### PR TITLE
Add perl files of openbmc2 to support openbmc on python

### PR DESCRIPTION
**Pull request by xuweibj**
1. Add openbmc2.pm

2. In openbmc.pm, check environment variable "XCAT_OPENBMC_PYTHON". If it is "YES", return.

3. In openbmc2.pm, when XCAT_OPENBMC_PYTHON is YES if there is no "/opt/xcat/lib/python/agent/agent.py", output as below:

```
# rpower f6u17 stat
Error: Failed to start python agent

# XCATBYPASS=1 rpower f6u17 stat
Error: '/opt/xcat/lib/python/agent/agent.py' does not exist
Error: Failed to start python agent
```

### CODE REVIEW

File: xCAT-server/lib/perl/xCAT/OPENBMC.pm [hunk 1]
```diff
@@ -43,4 +57,140 @@ sub send_request {
...
+    # always create a new lock file
+    unlink($LOCK_PATH);
+    open($lock_fd, ">>", $LOCK_PATH) or return undef;
+    flock($lock_fd, LOCK_EX) or return undef;
+    return $lock_fd;
+}
+sub start_python_agent {
+    my $agent_file = "/opt/xcat/lib/python/agent/agent.py";
+    if (! -e $agent_file) {
+        xCAT::MsgUtils->message("S", "Error: '/opt/xcat/lib/python/agent/agent.py' does not exist");
```
Comment (zet809) [inline xCAT-server/lib/perl/xCAT/OPENBMC.pm]: Use $agent_file is more flexible.

File: xCAT-server/lib/perl/xCAT/OPENBMC.pm [hunk 2]
```diff
@@ -43,4 +57,140 @@ sub send_request {
...
+        return;
+    }
+    $ret = $sock->send($buf);
+    if (!$ret) {
+        xCAT::MsgUtils->message("E", { data => ["Failed to send message to the agent"] }, $callback);
+        $sock->close();
+        kill('TERM', $pid);
+        return;
+    }
+    while(1) {
```
Comment (zet809) [inline xCAT-server/lib/perl/xCAT/OPENBMC.pm]: Do we need to add some timeout checking to exit the loop?

File: xCAT-server/lib/xcat/plugins/openbmc2.pm [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+=cut
+
+#-------------------------------------------------------
+sub process_request {
+    my $request = shift;
+    $callback = shift;
+    my $noderange = $request->{node};
+    my $check = parse_node_info($noderange);
+    $callback->({ errorcode => [$check] }) if ($check);
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/openbmc2.pm]: Do we need to check the content of %node_info before go to next step?
If nothing exists, we can exit directly.

File: xCAT-server/lib/xcat/plugins/openbmc2.pm [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    if (ref($request->{environment}) eq 'ARRAY' and ref($request->{environment}->[0]->{XCAT_OPENBMC_PYTHON}) eq 'ARRAY') {
+        $::OPENBMC_PYTHON = $request->{environment}->[0]->{XCAT_OPENBMC_PYTHON}->[0];
+    } elsif (ref($request->{environment}) eq 'ARRAY') {
+        $::OPENBMC_PYTHON = $request->{environment}->[0]->{XCAT_OPENBMC_PYTHON};
+    } else {
+        $::OPENBMC_PYTHON = $request->{environment}->{XCAT_OPENBMC_PYTHON};
+    }
+    
+    if (! (defined($::OPENBMC_PYTHON) and $::OPENBMC_PYTHON eq "YES")) {
+        $request = {};
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/openbmc2.pm]: If YES, we shall continue, right?

File: xCAT-server/lib/xcat/plugins/openbmc2.pm (hunk 3 above)
Comment (xuweibj) [reply xCAT-server/lib/xcat/plugins/openbmc2.pm]: Have added.

**Comment by zet809** [comment]
Looks good to me, I agree to merge, thx @xuweibj

**Pull request merged**